soda_machine.rb

```diff
@@ -1,18 +1,39 @@
+require_relative 'soda'
+
 class SodaMachine
   attr_reader :sodas, :cash
+  attr_accessor :current_inventory_count
+  
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
-  end
-
-  def current_inventory_count
+    @current_inventory_count = @sodas.length
   end
 
   def find_soda(soda_brand)
+    i = 0
+    while i < @sodas.length
+      if @sodas[i].brand == soda_brand 
+        @sodas[i]
+        break
+      end
+      i += 1
+    end
+    
+    @sodas[i]
+    # returns either match or nil 
   end
 
   def sell(soda_brand)
+    soda_to_sell = find_soda(soda_brand)
+
+    if soda_to_sell != nil
```

Comment:
No need to check it against `nil` -- `nil` is a falsey value, meaning that it will fail an `if` check, so you can just do `if soda_to_sell` and if it's `nil` it'll fail, and it'll pass otherwise (any truthy value, meaning anything other than `false` and `nil`, will pass an if check)